### PR TITLE
Making it possible to add files to strings

### PR DESCRIPTION
**Pull request by scottfrazer**
So users can do `${"--flag " + file_var}`.

I'm not totally sure what the semantics should be, but I think this will at least fix this use case decently well.

### CODE REVIEW

**Comment by mcovarr** [comment]
:+1:

File: src/main/scala/cromwell/binding/types/WdlStringType.scala [hunk 1]
```diff
@@ -21,6 +21,7 @@ case object WdlStringType extends WdlPrimitiveType {
 
   override def add(rhs: WdlType): Try[WdlType] = rhs match {
     case WdlStringType | WdlIntegerType | WdlFloatType => Success(WdlStringType)
+    case WdlFileType => Success(WdlFileType)
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/binding/types/WdlStringType.scala]: This generates a WdlString presumably?

"--file=/bin/bash" is not a valid path?

Comment (scottfrazer) [inline src/main/scala/cromwell/binding/types/WdlStringType.scala]: Yeah, I suppose String + File probably make the most sense as evaluating to a String... I can change

**Comment by cjllanwarne** [comment]
I'm not sure that adding a string to the beginning of a file name makes much sense - for your example wouldn't you just use "./script.sh --file=${file_variable}". And since WdlFile.toString is always fully qualified, you'll never get a useful WdlFile by prefixing a string to it?

**Comment by scottfrazer** [comment]
actually ` --file=${file_variable}` is different from `${"--file=" + file_variable}` subtly

In the latter case, file_variable can be optional, in the former case, it is required.

**Comment by cjllanwarne** [comment]
Oh, this is useful if you want to return some string followed by the filename. I agree WdlString + WdlFile should go to WdlString.

File: src/main/scala/cromwell/binding/values/WdlString.scala [hunk 2]
```diff
@@ -8,24 +8,25 @@ case class WdlString(value: String) extends WdlPrimitive {
   val wdlType = WdlStringType
 
   override def add(rhs: WdlValue): Try[WdlValue] = rhs match {
-    case r:WdlString => Success(WdlString(value + r.value))
-    case r:WdlInteger => Success(WdlString(value + r.value))
-    case r:WdlFloat => Success(WdlString(value + r.value))
+    case r: WdlString => Success(WdlString(value + r.value))
+    case r: WdlInteger => Success(WdlString(value + r.value))
+    case r: WdlFloat => Success(WdlString(value + r.value))
+    case r: WdlFile => Success(WdlFile(value + r.value))
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/binding/values/WdlString.scala]: And here.

**Comment by cjllanwarne** [comment]
:+1:

**Pull request merged**